entities/src/main/java/org/odk/collect/entities/EntitiesViewModel.kt

```diff
@@ -0,0 +1,37 @@
+package org.odk.collect.entities
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import org.odk.collect.async.Scheduler
+
+class EntitiesViewModel(
+    private val scheduler: Scheduler,
+    private val entitiesRepository: EntitiesRepository
+) : ViewModel() {
+
+    private val _datasets = MutableLiveData<List<String>>(emptyList())
+    val datasets: LiveData<List<String>> = _datasets
+
+    init {
+        scheduler.immediate(background = true) {
+            _datasets.postValue(entitiesRepository.getDatasets().toList())
+        }
+    }
+
+    fun getEntities(dataset: String): LiveData<List<Entity>> {
```

Comment:
`getEntities` is a method because the result needs an input (we filter on `dataset`) whereas the datasets does not need any input to be determined so it can just be a property.